### PR TITLE
Aleksei/send statistics to api

### PR DESCRIPTION
**Pull request by iambeone**
Closes #ISSUE

**Description:**

Sending stat data to the API

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/components/common/TmBalance.vue [hunk 1]
```diff
@@ -358,6 +358,7 @@ export default {
           )
           this.sentToGA = true
         }
+        // sending to ga only once
```
Comment (jbibla) [inline src/components/common/TmBalance.vue]: 1. are we still sending to GA?
2. is this comment in the right place? shouldn't it be on line 359?

File: tests/unit/specs/scripts/db_store.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+              action: "Action",
+              label: "Label",
+              value: "Value",
+              browserid: "browserid-uuid"
+            }
+          }
+        })
+      }
+      store_in_db(customObject, `Category`, `Action`, `Label`, `Value`)
+      expect(fetch).toHaveBeenCalledWith(`api/stat/store`, options)
```
Comment (jbibla) [inline tests/unit/specs/scripts/db_store.spec.js]: based on proposed changes in the API: https://github.com/luniehq/lunie-api/pull/359
```suggestion
      expect(fetch).toHaveBeenCalledWith(`api/stats/store`, options)
```

File: src/components/common/TmBalance.vue (hunk 1 above)
Comment (iambeone) [reply src/components/common/TmBalance.vue]: 1. Why not? )
2. Ooops

We can do it in API only, but without browser-id

File: src/scripts/db_store.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import browserID from "browser-id"
```
Comment (faboweb) [inline src/scripts/db_store.js]: move this logic to API to reduce surface?

File: src/scripts/google-analytics.js [hunk 4]
```diff
@@ -46,9 +47,11 @@ function customToNum(custom) {
   return false
 }
 
-module.exports.sendEvent = function event(customObject, ...args) {
+export const sendEvent = function event(customObject, ...args) {
   if (window.ga) {
     let newKey
+    // sending data before converting
+    store_in_db(customObject, ...args)
```
Comment (faboweb) [inline src/scripts/google-analytics.js]: nice way to reuse the existing logic 👍

File: src/scripts/db_store.js (hunk 3 above)
Comment (faboweb) [reply src/scripts/db_store.js]: clarification: proxy this request through the API as the API already is connected to the DB. we don't need to expose the DB to the FE.

File: src/scripts/db_store.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { getGraphqlHost } from "scripts/url"
+
+export const store_in_db = (customObject, ...args) => {
```
Comment (faboweb) [inline src/scripts/db_store.js]:
```suggestion
export const uploadStatistics = (customObject, ...args) => {
```

Also: let's rename the file to something similar like `upload_statics.js`

File: package.json [hunk 6]
```diff
@@ -45,6 +45,7 @@
     "axios": "^0.19.0",
     "bech32": "^1.1.3",
     "bignumber.js": "^8.1.1",
+    "browser-id": "1.1.0",
```
Comment (faboweb) [inline package.json]: this repo has 0 stars

Comment (faboweb) [reply package.json]: let's find an alternative or implement the logic ourselves

**Comment by faboweb** [comment]
How about:
Send a browser ID (maybe GA ID) as a header in GQL requests.
API requests trigger analytics writing.
Problem is only with tx volume? We could handle that case by adding meta data to our broadcasting requests.

**Comment by faboweb** [comment]
How about:
Send a browser ID (maybe GA ID) as a header in GQL requests.
API requests trigger analytics writing.
Problem is only with tx volume? We could handle that case by adding meta data to our broadcasting requests.

**Pull request closed without merging**